images/recycler/scripts/recycler.sh

```diff
@@ -22,8 +21,10 @@ if [[ ! -d ${dir} ]]; then
     exit 1
 fi
 
-# shred all files
-find ${dir} -type f -exec shred -fuvz {} \;
+find ${dir} -type f -exec stat -c "%u %g %n"  {} \; |xargs -n 3 bash -c 'sudo -u "#$1" shred  $3' argv0
```

Comment:
are you confident `find` will be able to recursively list all directories, even with the most restrictive permissions for another uid/gid? is there any way a user or group could create a directory tree that uid=0 inside the container could not find files in?

same question about `stat`... will it be able to read file uid/gid, even with the most restrictive permissions possible on them?